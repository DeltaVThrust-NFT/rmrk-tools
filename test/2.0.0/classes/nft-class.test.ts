import { createNftClassMock, addChangeIssuerMock, getBobKey } from "../mocks";
import { cryptoWaitReady } from "@polkadot/util-crypto";

beforeAll(async () => {
  return await cryptoWaitReady();
});

describe("rmrk2.0.0 NftClass: Create", () => {
  it("should match snapshot", async () => {
    const nftClass = createNftClassMock();
    expect(await nftClass.create()).toMatchSnapshot();
  });
});

describe("rmrk2.0.0 NftClass: Create", () => {
  it("should throw error", async () => {
    const nftClass = createNftClassMock(1);
    try {
      await nftClass.create();
    } catch (e) {
      expect(e.message).toMatch(
        "An already existing nft class cannot be created!"
      );
    }
  });
});

describe("rmrk2.0.0 NftClass: Change issuer", () => {
  it("should match snapshot", async () => {
    const nftClass = createNftClassMock(1);
    expect(await nftClass.change_issuer(getBobKey().address)).toMatchSnapshot();
  });
});

describe("rmrk2.0.0 NftClass: Create", () => {
  it("should throw error", async () => {
    const nftClass = createNftClassMock(0);
    try {
      await nftClass.change_issuer(getBobKey().address);
    } catch (e) {
      expect(e.message).toMatch(
        "This nft class is new, so there's no issuer to change. If it has been deployed on chain, load the existing nft class as a new instance first, then change issuer."
      );
    }
  });
});

describe("rmrk2.0.0 NftClass: Add change", () => {
  it("should match snapshot", async () => {
    const nftClass = createNftClassMock();
    expect(await nftClass.addChange(addChangeIssuerMock)).toMatchSnapshot();
  });
});

describe("rmrk2.0.0 NftClass: Get changes", () => {
  it("should match snapshot", async () => {
    const nftClass = createNftClassMock();
    await nftClass.addChange(addChangeIssuerMock);
    expect(await nftClass.getChanges()).toMatchSnapshot();
  });
});
