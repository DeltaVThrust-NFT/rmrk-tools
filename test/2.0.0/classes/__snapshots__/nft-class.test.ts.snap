// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rmrk2.0.0 NftClass: Change issuer should match snapshot 1`] = `"RMRK::CHANGEISSUER::2.0.0::d43593c715a56da27d-KANARIABIRDS::new_address"`;

exports[`rmrk2.0.0 NftClass: Create should match snapshot 1`] = `"RMRK::CREATE::2.0.0::%7B%22max%22%3A0%2C%22issuer%22%3A%225GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY%22%2C%22symbol%22%3A%22KANARIABIRDS%22%2C%22id%22%3A%22d43593c715a56da27d-KANARIABIRDS%22%2C%22metadata%22%3A%22https%3A%2F%2Fsome.url%22%7D"`;

exports[`rmrk2.0.0 NftClass: Get changes should match snapshot 1`] = `Array []`;
